src/main/kotlin/com/terraformation/backend/nursery/api/WithdrawalsController.kt

```diff
@@ -148,6 +151,21 @@ class WithdrawalsController(
 
     return ListWithdrawalPhotosResponsePayload(fileIds.map { NurseryWithdrawalPhotoPayload(it) })
   }
+
+  @ApiResponse200
+  @ApiResponse404("The withdrawal does not exist.")
+  @ApiResponse409("The withdrawal has already been undone.")
+  @PostMapping("/{withdrawalId}/undo")
+  @Operation(
+      summary = "Undoes a withdrawal.",
+      description =
+          "The withdrawal's plants will be returned to their original batches. Nursery transfers " +
+              "may not be undone. If the withdrawal was an outplanting to a planting site, the " +
```

Comment:
Will the 409 CONFLICT also handle the trying to undo the nursery transfer?

Reply:
Good catch; I updated the description to be more general.